### PR TITLE
Add optional commands

### PR DESCRIPTION
**Pull request by KazuCocoa**
related to: https://github.com/appium/appium-doctor/pull/48

```
$ ./appium-doctor.js
info AppiumDoctor Appium Doctor v.1.6.0
info AppiumDoctor ### Diagnostic for necessary dependencies starting ###
info AppiumDoctor  ✔ The Node.js binary was found at: /Users/kazuaki/.nodebrew/current/bin/node
info AppiumDoctor  ✔ Node version is 10.11.0
info AppiumDoctor  ✔ Xcode is installed at: /Applications/Xcode.app/Contents/Developer
info AppiumDoctor  ✔ Xcode Command Line Tools are installed in: /Applications/Xcode.app/Contents/Developer
info AppiumDoctor  ✔ DevToolsSecurity is enabled.
info AppiumDoctor  ✔ The Authorization DB is set up properly.
WARN AppiumDoctor  ✖ Carthage was NOT found!
info AppiumDoctor  ✔ HOME is set to: /Users/kazuaki
info AppiumDoctor  ✔ ANDROID_HOME is set to: /Users/kazuaki/Library/Android/sdk
info AppiumDoctor  ✔ JAVA_HOME is set to: /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home
info AppiumDoctor  ✔ adb exists at: /Users/kazuaki/Library/Android/sdk/platform-tools/adb
info AppiumDoctor  ✔ android exists at: /Users/kazuaki/Library/Android/sdk/tools/android
info AppiumDoctor  ✔ emulator exists at: /Users/kazuaki/Library/Android/sdk/tools/emulator
info AppiumDoctor  ✔ Bin directory of $JAVA_HOME is set
info AppiumDoctor ### Diagnostic for necessary dependencies completed, one fix needed. ###
info AppiumDoctor
info AppiumDoctor ### Diagnostic for optional dependencies starting ###
info AppiumDoctor  ✔ opencv4nodejs is installed.
WARN AppiumDoctor  ✖ fbsimctl cannot be found
WARN AppiumDoctor  ✖ applesimutils cannot be found
info AppiumDoctor ### Diagnostic for optional dependencies completed, 2 fixes needed. ###
info AppiumDoctor
info AppiumDoctor ### Manual Fixes Needed ###
info AppiumDoctor The configuration cannot be automatically fixed, please do the following first:
WARN AppiumDoctor  ➜ Please install Carthage. Visit https://github.com/Carthage/Carthage#installing-carthage for more information.
info AppiumDoctor
info AppiumDoctor ### Optional Manual Fixes ###
info AppiumDoctor The configuration can install optionally. Please do the following by manual:
WARN AppiumDoctor  ➜ Why fbsimctl needs and how to install it is: http://appium.io/docs/en/drivers/ios-xcuitest/
info AppiumDoctor
info AppiumDoctor ###
info AppiumDoctor
info AppiumDoctor Bye! Run appium-doctor again when all manual fixes have been applied!
info AppiumDoctor
```

image

What about this style?

- necessary commands
    - ok, nOk, manual fix, autofix
- optional commands
    - okOptional, nOkOptional, manual fix (no autofix)

I separated sections for necessary & optional

---

~~I would clean implementation up more. First, I would like to ensure the descriptions of output.~~

### CODE REVIEW

File: lib/ios.js [hunk 1]
```diff
@@ -142,8 +142,34 @@ checks.push(new CarthageCheck());
...
+
+class OptionalApplesimutilsCommandCheck extends DoctorCheck {
+  async diagnose () {
+    const applesimutilsPath = await resolveExecutablePath('applesimutils');
+    return applesimutilsPath
+      ? okOptional(`applesimutils is installed at: ${applesimutilsPath}. Installed versions are: ${(await exec('brew', ['list', '--versions', 'applesimutils'])).stdout.trim()}`)
+      : nokOptional('applesimutils cannot be found');
+  }
+  async fix () { // eslint-disable-line require-await
+    return `Why fbsimctl needs and how to install it is: http://appium.io/docs/en/drivers/ios-xcuitest/`;
```
Comment (imurchie) [inline lib/ios.js]: "Why applesimutils is needed and how to install:"

**Comment by KazuCocoa** [comment]
fixed wording & wrapped up code base

**Comment by KazuCocoa** [comment]
Could you release the new version when you have a change?
(cc @imurchie @jlipps )

Mainly, the next release will have:
1. Fix `node` check in Windows #45
2. Fix Xcode CLI check for macOS 10.11+ #44
3. Add hins for optional commands (this PR)

**Comment by imurchie** [comment]
@KazuCocoa `appium-doctor@1.7.0` is out.

**Pull request merged**